wagtail/admin/templates/wagtailadmin/shared/dialog/dialog.html

```diff
@@ -0,0 +1,44 @@
+{% load wagtailadmin_tags i18n %}
+{% with id=title|cautious_slugify %}
+    <template data-dialog>
+        <div
+            id="{{ id }}"
+            aria-labelledby="title-{{ id }}"
+            aria-hidden="true"
+            class="{{ class_name }} dialog"
+        >
+            <div data-a11y-dialog-hide class="dialog__overlay"></div>
+            <div class="dialog__box">
+                <button type="button" data-a11y-dialog-hide aria-label="{% trans 'Close dialog' %}" class="dialog__close-button">
+                    {% icon name='cross' class_name="dialog__close-icon" %}
+                </button>
+
+                {# Message #}
+                {% if message_heading and message_icon_name %}
+                    <div class="dialog__message dialog__message--{{ message_status }}">
+                        {% icon name=message_icon_name class_name="dialog__message-icon "|add:message_icon_color %}
+                        <div class="dialog__message-header">
+                            <strong class="dialog__message-heading">{{ message_heading }}</strong>
+                            {% if message_description %}<p class="dialog__message-description ">{{ message_description }}</p>{% endif %}
+                        </div>
+                    </div>
+                {% endif %}
+
+                <div class="dialog__content">
+                    {# Title #}
+                    <h2 class="dialog__title" id="title-{{ id }}">
+                        {% if icon_name %}{% icon name=icon_name class_name="dialog__icon" %}{% endif %}
+                        {{ title }}
+                    </h2>
+
+                    {# Subtitle #}
```

Comment:
Double checking - is this going to change back to description?

Reply:
I'm quite keen on keeping it as a subtitle or perhaps we could rename this and the title above to: `heading / subheading` because thats what it more accurately represents in the visual styling of this component.  
### screenshot for visual example 
![Screen Shot 2022-05-13 at 4 06 13 PM](https://user-images.githubusercontent.com/25041665/168395408-1efbcdf5-29e6-4b52-9d02-1f348085a2a5.png)
 
I'm a little lost on what value it would add to have: `<h2>{{ title }}{{ subtitle}}</h2>` vs `<h2>{{ title }}</h2>` in the future, since they would render the same by passing your entire string (title & subtitle) to {{ title }}. 

Unless you're suggesting that title && subtitle both reside inside of the h2 but are styled similar to the screenshot above so that it all registers as 1 heading to assistive tech?

Sorry for the confusion too

Reply:
My reasoning is the previous convention set by the header template. 

See this PR for context. 
https://github.com/wagtail/wagtail/pull/8368

- if we have a subtitle - I think it should reside within the h2 in a span, it is part of the title just visually shown different
- we should keep the original description content and have this sit in a div outside of the header. P tag is also good

Reply:
Happy with heading/subheading but just trying to avoid too much different naming of things across similar (conceptually) content. 

But I understand that the modal is not a page and hence the title/subtitle could be reserved for that only. 

I know these are nitpicky but just trying to look at the conventions we have set elsewhere and try to honour them in new code (where it makes sense).

Reply:
> * if we have a subtitle - I think it should reside within the h2 in a span, it is part of the title just visually shown different


Awesome! Thanks for the clarification. I have moved this to reside inside of the h2 to bring it more in sync with the other title/subtitle combos we have.

Reply:
Thank you. This is much cleaner.